src/date/start-of.js

```diff
@@ -7,17 +9,21 @@ define(function() {
  * seconds and milliseconds to 0. Starting at the month does the same, but
  * also sets the date to 1.
  *
+ * calendar is the name of the calendar system, to determine what a "year" and "month" are
+ *
  * Returns the modified date
  */
-return function( date, unit ) {
-	date = new Date( date.getTime() );
+return function( date, unit, calendar ) {
+	if ( unit === "year" ) {
+		date = new gdateGdate.calendars[ calendar ]( date ).startOfYear().toDate();
+	} else if ( unit === "month" ) {
+		date = new gdateGdate.calendars[ calendar ]( date ).startOfMonth().toDate();
+	} else {
+		date = new Date( date.getTime() );
+	}
 	switch ( unit ) {
 		case "year":
-			date.setMonth( 0 );
-		/* falls through */
 		case "month":
-			date.setDate( 1 );
-		/* falls through */
 		case "day":
 			date.setHours( 0 );
 		/* falls through */
```

Comment:
I don't think we need the startOf function at all. I haven't eliminated it yet, but it should be.